packages/@o3r/telemetry/src/builders/index.ts

```diff
@@ -31,32 +32,37 @@ export const createBuilderWithMetrics: BuilderWrapper = (builderFn) =>
       throw (e instanceof Error ? e : new Error(error));
     }
     finally {
-      const endTime = Math.floor(performance.now());
-      const duration = endTime - startTime;
-      // context.builder.builderName does not contain the package name
-      const builderName = context.builder.name as string;
-      context.logger.info(`${builderName} run in ${duration}ms`);
-      const environment = getEnvironmentInfo();
-      const data = {
-        environment,
-        duration,
-        builder: {
-          name: builderName,
-          options,
-          ...(context.target
-            ? {
-              target: {
-                name: context.target.target,
-                projectName: context.target.project,
-                configuration: context.target.configuration
-              }
-            } : {}
-          )
-        },
-        error
-      };
-      context.logger.debug(JSON.stringify(data, null, 2));
-      // TODO handle call to log server here https://github.com/AmadeusITGroup/otter/issues/1201
+      try {
+        const endTime = Math.floor(performance.now());
+        const duration = endTime - startTime;
+        // context.builder.builderName does not contain the package name
+        const builderName = context.builder.name as string;
+        context.logger.info(`${builderName} run in ${duration}ms`);
+        const environment = getEnvironmentInfo();
+        const data: BuilderMetricData = {
+          environment,
+          duration,
+          builder: {
+            name: builderName,
+            options,
+            ...(context.target
+              ? {
+                target: {
+                  name: context.target.target,
+                  projectName: context.target.project,
+                  configuration: context.target.configuration
+                }
+              } : {}
+            )
+          },
+          error
+        };
+        context.logger.debug(JSON.stringify(data, null, 2));
+        await sendData(data, context.logger);
+      } catch (e: any) {
+        // Do not throw error if we don't manage to collect data
+        context.logger.error(e.toString());
```

Comment:
we loose the stack trace by doing `toString`, if the logger doesn't support Error object, we should build a string containing `e.message` and `e.stack`

Reply:
![image](https://github.com/AmadeusITGroup/otter/assets/52541061/ea16d285-3da5-4d65-a763-cd3a448e7e75)
I set `e.stack || e.toString()`